### PR TITLE
[VIP][Travel] Open travelDot after accepting terms and conditions

### PR DESCRIPTION
**Pull request by rushatgabhane**
### Fixed Issues

$ https://github.com/Expensify/App/issues/43867
PROPOSAL: https://expensify.slack.com/archives/C05S5EV2JTX/p1718718276561529?thread_ts=1718697839.930099&cid=C05S5EV2JTX

### Tests

Pre-requisite: Account should be on travel beta.

#### Travel terms of service not accepted by the account

1. Go to an account that hasn't accepted travel Terms of Service (ToS) before.
2. Open global `+` > Book travel
3. Click book travel.
4. Verify that you're shown (ToS) page.
5. Accept the ToS and click next.
6. Verify that you're redirected to travelDot.

#### Travel terms of service previously accepted by the account
1. Go to an account that has accepted the above mentioned ToS.
2. Open global `+` > Book travel
3. Click book travel.
4. Verify that you're not shown the ToS page.
5. Verify that you're redirected to travelDot.

- [x] Verify that no errors appear in the JS console

### QA Steps

Same as Tests
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/29673073/d2e569d5-8bf2-4dae-93b9-6bc5c35a62f9

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/29673073/2306319a-2ce3-4f3c-9a1a-876c59ca29b8

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/29673073/4cf18268-d8e6-4777-aaa2-03d15df4c786

https://github.com/Expensify/App/assets/29673073/a3b074a1-40f1-4bb1-8e78-3856453f80b5

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

### ToS Not accepted
https://github.com/Expensify/App/assets/29673073/4039de5c-8241-4186-95d9-4c21c4d0fecf

### ToS accepted
https://github.com/Expensify/App/assets/29673073/c4fc6ffd-4746-47e7-8f33-cdb6017ce5ce

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/29673073/895d22fe-b39c-49d1-bbbb-024c0c88e317

</details>

### CODE REVIEW

File: src/libs/actions/Travel.ts [hunk 1]
```diff
@@ -1,25 +1,31 @@
...
             },
         },
     ];
 
-    API.write(WRITE_COMMANDS.ACCEPT_SPOTNANA_TERMS, {}, {successData});
+    asyncOpenURL(
+        // eslint-disable-next-line rulesdir/no-api-side-effects-method
+        API.makeRequestWithSideEffects(SIDE_EFFECT_REQUEST_COMMANDS.ACCEPT_SPOTNANA_TERMS, {}, {optimisticData})
+            .then((response) => (response?.spotnanaToken ? buildTravelDotURL(response.spotnanaToken) : buildTravelDotURL()))
+            .catch(() => buildTravelDotURL()),
```
Comment (rushatgabhane) [inline src/libs/actions/Travel.ts:26]: Error handled will be done in https://github.com/Expensify/App/issues/43780

**Comment by rushatgabhane** [comment]
cc @stitesExpensify @danieldoglas : )

**Comment by rushatgabhane** [comment]
screenshots for all platforms on the way

**Comment by jayeshmangwani** [comment]
Not sure if it needs a C+ review. Please let me know if it does.

**Comment by twisterdotcom** [comment]
Gonna assign @shubham1206agra here if that's okay @jayeshmangwani as he has a lot more context on travel. Also @danieldoglas too for an internal review.

**Comment by rushatgabhane** [comment]
The TS error is from `main`. Kindly ignore it

**Review by shubham1206agra — APPROVED** [review]

**Comment by rushatgabhane** [comment]
@danieldoglas this can be merged if it looks good to you

**Review by danieldoglas — APPROVED** [review]

**Pull request merged**